### ISSUE TITLE
Installation of GithubDesktop 2.5.6 for Windows fails

### ISSUE THREAD

**Issue by mich123prem** [desktop/desktop#10961]
### Describe the bug

I start the installation process as administrator (as I have always previewsly used to do).
When the installation is complete, I get the following message:
![bilde](https://user-images.githubusercontent.com/68922815/98014044-6d23ed80-1dfb-11eb-89d5-47e6b517b427.png)

### Version & OS
Github desktop 2.5.6, 
OS-name	Microsoft Windows 10 Education
Version	10.0.18363 Build 18363
Prosessor	Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz, 2304 Mhz, 4 kjerne(r), 8 logiske prosessor(er)

Open 'About GitHub Desktop' menu to see the Desktop version. Also include what operating system you are using.

### Steps to reproduce the behavior

1. Go to 'clone a repository from the internet'
2. Click on 'one of the repos, and "clone"'
![bilde](https://user-images.githubusercontent.com/68922815/98015239-d9ebb780-1dfc-11eb-90aa-7113575932df.png)

### Expected behavior
Cloning of repository.

### Actual behavior

The above error message

### Screenshots

Add screenshots to help explain your problem, if applicable.

### Logs

Attach your logs by opening the `Help` menu and selecting `Show Logs...`, if applicable.
2020-11-03T13:53:09.659Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T13:53:10.041Z - info: [ui] launching: 2.5.6 (Windows 10.0.18363)
2020-11-03T13:53:10.041Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe'
2020-11-03T13:53:10.537Z - info: [ui] Stats reported.
2020-11-03T13:54:23.500Z - error: [ui] createRepository: unable to initialize a Git repository at C:\Users\michaelp\OneDrive - OsloMet\Nov20\GitHub\ElTeC
Error: Failed to execute initGitRepository: UNKNOWN
    at C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\lib\git\core.ts:143:13
    at Uo (C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\lib\git\core.ts:136:9)
    at C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\lib\git\init.ts:4:41
    at Object.Lv.createRepository [as onSubmit] (C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\ui\add-repository\create-repository.tsx:230:5)
2020-11-03T14:13:55.719Z - error: [ui] createRepository: unable to initialize a Git repository at C:\Users\michaelp\OneDrive - OsloMet\Nov20\GitHub\Eltec-NOR
Error: Failed to execute initGitRepository: UNKNOWN
    at C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\lib\git\core.ts:143:13
    at Uo (C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\lib\git\core.ts:136:9)
    at C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\lib\git\init.ts:4:41
    at Object.Lv.createRepository [as onSubmit] (C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\webpack:\app\src\ui\add-repository\create-repository.tsx:230:5)
2020-11-03T14:15:21.875Z - info: [main] Received possible protocol arguments: 3
2020-11-03T14:15:21.876Z - info: [main] Processing protocol url
2020-11-03T14:15:22.265Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T14:15:22.595Z - info: [ui] launching: 2.5.6 (Windows 10.0.18363)
2020-11-03T14:15:22.595Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe'
2020-11-03T14:15:22.624Z - info: [ui] [Dispatcher] requesting authenticated user
2020-11-03T14:15:22.624Z - warn: [ui] requestAuthenticatedUser was not called with valid OAuth state. This is likely due to a browser reloading the callback URL. Contact GitHub Support if you believe this is an error
2020-11-03T14:15:53.511Z - error: [ui] createRepository: unable to initialize a Git repository at C:\Users\michaelp\OneDrive - OsloMet\Dokumenter\GitHub\ElTeC-Nor
Error: Failed to execute initGitRepository: UNKNOWN
    at C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\lib\git\core.ts:143:13
    at Uo (C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\lib\git\core.ts:136:9)
    at C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\lib\git\init.ts:4:41
    at Object.Lv.createRepository [as onSubmit] (C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\ui\add-repository\create-repository.tsx:230:5)
2020-11-03T14:16:59.276Z - error: [ui] createRepository: unable to initialize a Git repository at C:\Users\michaelp\OneDrive - OsloMet\Dokumenter\GitHub\ElTeC-Nor
Error: Failed to execute initGitRepository: UNKNOWN
    at C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\lib\git\core.ts:143:13
    at Uo (C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\lib\git\core.ts:136:9)
    at C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\lib\git\init.ts:4:41
    at Object.Lv.createRepository [as onSubmit] (C:\Program Files (x86)\Microsoft\Edge\Application\86.0.622.58\webpack:\app\src\ui\add-repository\create-repository.tsx:230:5)
2020-11-03T14:20:41.756Z - info: [ui] [AppStore] adding account mich123prem (mich123prem) to store
2020-11-03T14:21:33.699Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T14:21:42.566Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T14:55:50.517Z - info: [main] Received possible protocol arguments: 2
2020-11-03T14:55:50.520Z - error: [main] Malformed launch arguments received: C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe,--squirrel-firstrun
2020-11-03T14:55:51.001Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T14:55:51.001Z - info: [ui] [AppStore] found account: mich123prem (mich123prem)
2020-11-03T14:55:51.591Z - info: [ui] launching: 2.5.6 (Windows 10.0.18363)
2020-11-03T14:55:51.592Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe'
2020-11-03T15:09:09.664Z - info: [main] Error report submitted
2020-11-03T15:10:11.726Z - info: [main] Received possible protocol arguments: 3
2020-11-03T15:10:11.727Z - info: [main] Processing protocol url
2020-11-03T15:10:12.084Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T15:10:12.084Z - info: [ui] [AppStore] found account: mich123prem (mich123prem)
2020-11-03T15:10:12.213Z - info: [ui] launching: 2.5.6 (Windows 10.0.18363)
2020-11-03T15:10:12.213Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe'
2020-11-03T15:10:12.238Z - info: [ui] [Dispatcher] requesting authenticated user
2020-11-03T15:10:12.238Z - warn: [ui] requestAuthenticatedUser was not called with valid OAuth state. This is likely due to a browser reloading the callback URL. Contact GitHub Support if you believe this is an error
2020-11-03T15:10:48.653Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:52.941Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:53.686Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:54.270Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:55.070Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:55.670Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:56.200Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:10:57.078Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:11:02.390Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:11:15.574Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:13:43.130Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:13:46.887Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:13:47.567Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:13:48.679Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:19:32.050Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:19:34.695Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:19:35.575Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:19:56.526Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:19:58.271Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T15:19:58.999Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)
2020-11-03T16:24:20.630Z - info: [main] Received possible protocol arguments: 2
2020-11-03T16:24:20.632Z - error: [main] Malformed launch arguments received: C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.4\GitHubDesktop.exe,--squirrel-firstrun
2020-11-03T16:24:21.467Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T16:24:21.467Z - info: [ui] [AppStore] found account: mich123prem (mich123prem)
2020-11-03T16:24:22.000Z - info: [ui] launching: 2.5.4 (Windows 10.0.18363)
2020-11-03T16:24:22.000Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.4\GitHubDesktop.exe'
2020-11-03T16:37:04.783Z - info: [main] Received possible protocol arguments: 2
2020-11-03T16:37:04.785Z - error: [main] Malformed launch arguments received: C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.4\GitHubDesktop.exe,--squirrel-firstrun
2020-11-03T16:37:05.430Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T16:37:05.430Z - info: [ui] [AppStore] found account: mich123prem (mich123prem)
2020-11-03T16:37:05.907Z - info: [ui] launching: 2.5.4 (Windows 10.0.18363)
2020-11-03T16:37:05.908Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.4\GitHubDesktop.exe'
2020-11-03T16:44:33.348Z - info: [main] Received possible protocol arguments: 2
2020-11-03T16:44:33.349Z - error: [main] Malformed launch arguments received: C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe,--squirrel-firstrun
2020-11-03T16:44:34.218Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T16:44:34.218Z - info: [ui] [AppStore] found account: mich123prem (mich123prem)
2020-11-03T16:44:34.667Z - info: [ui] launching: 2.5.6 (Windows 10.0.18363)
2020-11-03T16:44:34.667Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe'
2020-11-03T16:46:23.656Z - info: [main] Received possible protocol arguments: 3
2020-11-03T16:46:23.657Z - info: [main] Processing protocol url
2020-11-03T16:46:24.026Z - info: [ui] [AppStore] loading 0 repositories from store
2020-11-03T16:46:24.026Z - info: [ui] [AppStore] found account: mich123prem (mich123prem)
2020-11-03T16:46:24.160Z - info: [ui] launching: 2.5.6 (Windows 10.0.18363)
2020-11-03T16:46:24.160Z - info: [ui] execPath: 'C:\Users\michaelp\AppData\Local\GitHubDesktop\app-2.5.6\GitHubDesktop.exe'
2020-11-03T16:46:24.182Z - info: [ui] [Dispatcher] requesting authenticated user
2020-11-03T16:46:24.183Z - warn: [ui] requestAuthenticatedUser was not called with valid OAuth state. This is likely due to a browser reloading the callback URL. Contact GitHub Support if you believe this is an error
2020-11-03T16:48:29.022Z - info: [ui] [AppStore.getAccountForRemoteURL] account found for remote: https://github.com/mich123prem/ELTeC-nor.git - mich123prem (has token)

**Comment by steveward** [comment]
@mich123prem thanks for the report. It looks like you are trying to create repositories in your OneDrive directory. Generally we recommend against creating repositories in file syncing directories since it can cause the repository to become corrupt. Can you try creating a repository in a different directory to see if you encounter this error there?

**Comment by mich123prem** [comment]
Thanks for a rapid response !!!

Yes, I will.
But, I already get an error message at installation time.
One drive repositories have not represented any problems in previous versions.

Michael

**Comment by mich123prem** [comment]
Tried now.
I get the same error also when cloning to a local directory.

Michael

**Comment by steveward** [comment]
I'd recommend trying a fresh installation:

1. Close GitHub Desktop
2. Remove all of the files in `%LocalAppData%\GitHubDesktop\`
3. Reinstall from http://desktop.github.com

Let me know if that helps.

**Comment by mich123prem** [comment]
I have done so several times. Re-startet and re-installed.
But:
The last version that functioned well has been 2.5.4.
Is there any possibility to get this version and install it? I cannot find it anywhere. Is it still available?

Michael

**Comment by Shaker4130** [comment]
Nadi

**Comment by steveward** [comment]
Thanks for the update @mich123prem. Are you running any antivirus/security software on this machine that could be interfering with the installation of GitHub Desktop? You mentioned needing to run as an administrator -- are there specific restrictions on this machine that require that?

**Comment by mich123prem** [comment]
Yes. This is a work PC (I work at a university), so I am installing with a limited administrator account.
But this was also the case for the previous version I was running.

**Comment by steveward** [comment]
I don't believe there were any changes that would have caused the application to break like this during update. There is a previous issue where a similar error was reported (https://github.com/desktop/desktop/issues/9223), and the cause there also seemed related to using a non-admin account. 

I also just tested this out in a VM where I had a non-admin account. I updated from v2.5.3 -> v2.5.7 and was unable to reproduce this error. Is there anyone at your university IT department who might have some insight into what could be causing this on your machine?

**Comment by mich123prem** [comment]
Thanks for your effort.
I will investigate this further with the system (not Github Desktop) as the main suspect.

Michael

**Comment by steveward** [comment]
@mich123prem let us know what you are able to find out and we can continue to dig into this. I'm going to close this issue out for the time being.

**Comment by mich123prem** [comment]
The problem seems to have been that Github Desktop installed itself into the AppData hierarchy.
Moving the entire directory to c:\ Program Files solved the problem.

Michael

**Comment by steveward** [comment]
@mich123prem GitHub Desktop installs by default to the AppData directory. Changing the installation directory is not recommended. Do you have specific restrictions in your environment against installing applications to AppData?

**Comment by mich123prem** [comment]
Not that i know of, but in AppData it caused problems (never actually properly starting), and under Program Files it (so far) works normally .

Michael

**Comment by kabepst** [comment]
@steveward I do know this is a old thread but I think I know where this comes from and would need your help in a similiar case.

This seems to be related to Windows AppLocker Rules. At my working place we had used Github Desktop for months without any problems (installed to %localappdata% via the Github Desktop MSI Installer for All-Users). Currently we are doing some QA-Tests for restricting Applications via Windows GPO-AppLocker Rules. What we tried:

1. Enable Default App-Locker Rules and update local policy on client (ignore Rules without "Default" in it's name) --> Screenshot 1 is from Executable Rules and Screenshot 2 from Packaged app Rules
![image](https://user-images.githubusercontent.com/70746076/233340227-bbc213a8-0cdb-4fa4-ac4d-59b49d7d8ca3.png)
![image](https://user-images.githubusercontent.com/70746076/233340417-d4294a10-d30a-4e5f-aedc-a53da76f54fa.png)

2. Try to run Github Desktop: Will be blocked by AppLocker as expected
3. Add the "Allow Github Desktop" Rule you already have seen above. This rule is a "Publisher based" Conditional Rule:
![image](https://user-images.githubusercontent.com/70746076/233341823-a11382ad-de56-4388-9deb-9a605c70c349.png)

4. Update Policy on client and try to run Github Desktop again. Github Desktop will start, but the mentioned error ("Failed to execute installGlobalLFSFilter: UNKNOWN") popus. Also I'm not able to clone/push/pull from any repository if i click on Close in the error message afterwards.(Repositories are located in C:\Github\ in this example, no OneDrive or similiar things in place).
 

For me this looks like there is another binary or simliar that Github Desktop tries to run which is signed differently than the Github Desktop.exe itself (see the rule screenshot above) and Windows fails to display the default "This App is blocked by your Administrator" message as it would happen when running the exe directly without an AppLocker Exception. 

What could be the root cause and fix it:

- Is there a git.exe or simliar binary which will be called when git commands are running within the %localappdata%\GithubDesktop Folder? If so: Can you please provide me the correct paths to them? This would allow me to create yet another rule and allow the execution of this binary. 
- Or perhaps: Is there a "recommended" AppLocker Configuration from Microsoft itself for Github Desktop ? For example see this for Microsoft Teams: https://learn.microsoft.com/en-us/microsoftteams/applocker-in-teams

If you need any log files or similiar things for further analyse don't hesitate to get in touch with me. I'm happy to help to find the root cause of this and fix it :)

**Comment by kabepst** [comment]
@steveward 
So I've done another quick test and I'm quite sure it has to be the explained problem (running a binary which is signed by another  Publisher as GithubDesktop.exe). For testing purposes I've created a path-exception (allow everything within %localappdata%\GithubDesktop) and there we go: Github Desktops is once again working as expected. 

The rule itself:
![image](https://user-images.githubusercontent.com/70746076/233345703-cc8149d3-c1fc-4e52-b1cf-9d9f2bb6aec0.png)

However: 

- I would really like to avoid to have a "path-based" exception within the AppLocker-rules to a path which is "writeable" by the enduser (as people could use this folder to bypass AppLocker-Rules, for example: just copy a standalone binary to this folder and execute it..)
- I've done a quick search within the folders (*.exe) and there are plenty of them. So creating a rule for each of them isn't an option imho BUT: If there would be a list of all signed publishers (I hope there are only 2 or 3 ...)  for those exe's it could be an option (because of that I referenced the MS Teams example above as it seems that Microsoft does indeed provided such lists for other products, perhaps this is also possible for Github Desktop?)
- Another option would be (at least for me) to move the whole installation to C:\Program Files or similiar folder. But this does not seem to be supported by Github Desktop if I'm right?
